frontend/src/metabase/search/components/SearchFilterModal/SearchFilterModal/SearchFilterModal.tsx

```diff
@@ -29,16 +29,30 @@ export const SearchFilterModal = ({
 }) => {
   const [output, setOutput] = useState<SearchFilters>(value);
 
-  const filterMap: Record<FilterTypeKeys, SearchFilterComponent> = {
-    [SearchFilterKeys.Type]: TypeFilter,
-    [SearchFilterKeys.Verified]: PLUGIN_CONTENT_VERIFICATION.VerifiedFilter,
-  };
+  const filterMap: Record<FilterTypeKeys, SearchFilterComponent> = useMemo(
+    () => ({
+      [SearchFilterKeys.Type]: TypeFilter,
+      [SearchFilterKeys.Verified]: PLUGIN_CONTENT_VERIFICATION.VerifiedFilter,
+    }),
+    [],
+  );
+
+  const isValidFilterValue = useCallback(
+    (
+      key: FilterTypeKeys,
+      val: SearchFilterPropTypes[FilterTypeKeys],
+    ): boolean =>
+      !!val &&
+      (!Array.isArray(val) || val.length > 0) &&
+      !!filterMap[key as FilterTypeKeys],
```

Comment:
This will prevent any possibility of `verified: true` appearing in the filter object if the user access the URL `/search?verified=true` in a non-premium instance

Reply:
I think overall, this function is too hard to follow, can you simplify this more? You don't need to just cramp everything into a single expression. Using a longer-body function with lots of return is okay as long as it's readable.